convert2rhel/unit_tests/README.md

```diff
@@ -0,0 +1,170 @@
+    monkeypatch.setattr(package_handler, "get_packages_to_update", value=packages_to_update_mock)
+
+    assert package_handler.get_packages_to_update() is not None
+    assert packages_to_update_mock.assert_called_once_with(["package-1", "package-2"])
+```
+
+And this other example, of a test that don't need any mocks for external
+dependencies:
+
+```python
+import os
+import file_handler
+
+def test_archive_old_files(tmpdir):
+    tmpdir = str(tmpdir)
+    some_dir = os.path.join(tmpdir, "some_dir")
+    some_file = os.path.join(tmpdir, "some_file.txt")
+    test_data = "test data\n"
+
+    open(some_file, "w").write(test_data)
+
+    assert "some_file.txt" in os.listdir(tmpdir)
+
+    file_handler.archive_old_files(tmpdir)
+
+    old_files = os.listdir(some_dir)
+    assert len(old_files) == 1
+    with open(os.path.join(some_dir, old_files[0])) as old_f:
+        assert old_f.read() == test_data
+```
+
+The difference between the two is that, the first one we can't rely on the
+output of the `package_handler` (Because in every test run, the result may be
+different depending on the system, because we are relying on an external
+dependency), so we mock it to be sure we have full control over the results,
+and for the second one, we know it's such a simple function that just move old
+files to another directory, so it's better to provide the files and some tmp
+path, still having full control of the output, no mocking necessary.
+
+## Running the tests
+
+The tests can be run in two different ways, `Locally` or with `Docker`.
+
+Running the tests locally has the advantage of using a debugger to follow the
+execution of the code, and it's easier to generate coverage reports from the
+tests.
+
+In the other hand, running the tests in a container guarantees that you have a
+isolated running test case, without interference from your machine.
+
+### Locally
+
+To run the tests locally, you can use an alteady made `make` command we have
+defined in our `Makefile`, which is:
+
+```bash
+# Make command to run the tests with pytest
+make tests-locally
+```
+
+Running the tests locally also allow you to execute other commands from pytest,
+such as:
+
+```bash
+# Run pytest and gather code coverage
+pytest --cov
+
+# Run a specific test (or group of tests based on name)
+pytest -k test_my_thing
+```
+
+### Docker
+
+To run the tests using a docker container, you can do so with the below command.
+
+This command will run the full tests suit in both CentOS 7 and 8.
+
+```bash
+# Running tests in both CentOS 7 and 8
+make tests
+```
+
+And, if you want to run the tests in versions individually, you can just do
+this:
+
+```bash
+# Running the tests only in CentOS 7
+make tests7
+
+# Running the tests only in CentOS 8
+make tests8
+```
+
+## Checking for test coverage
+
+After running the tests, you can check for the coverage in a nicer HTML
+visualization instead of the terminal output.
+
+For this, run those command in your terminal:
```

Comment:
We'd need to run `pytest --cov` to get the coverage report right?

Reply:
You're right, I forgot to add the statement to run pytest with --cov enabled.